extensions/amp-auto-ads/0.1/placement.js

```diff
@@ -187,21 +189,16 @@ export class Placement {
   }
 
   /**
-   * @param {string} type
-   * @param {!Array<!DataAttributeDef>} dataAttributes
+   * @param {!Object<string, string>} baseAttributes
    * @return {!Element}
    * @private
    */
-  createAdElement_(type, dataAttributes) {
-    const attributes = {
-      type,
+  createAdElement_(baseAttributes) {
+    const attributes = Object.assign({
       'layout': 'responsive',
       'width': '0',
       'height': '0',
-    };
-    for (let i = 0; i < dataAttributes.length; ++i) {
-      attributes['data-' + dataAttributes[i].name] = dataAttributes[i].value;
-    }
+    }, baseAttributes, this.attributes_);
```

Comment:
See attributes.js, they all get parsed through there which ensures they either begin with data- or are in a strict whitelist.

I can make this extra safe by having attributes be contained within a dedicated Attributes class, to help prevent someone adding a vulnerability in the future.